### ISSUE TITLE
Is it possible to separate photo storage (library) and thumbs/encoded-video 

### ISSUE THREAD

**Issue by stevetsanders** [imagegenius/docker-immich#437]
I'm wondering if it's possible to separate the photo storage location from the generated thumbnails/encoded-video location. I've seen some posts about separating them to take advantage of unraid parity for the actual photos but keep the thumbnails/encoded-video/other stuff on the cache. Ideally the setup would this I think

unraid array: 
- library folder

unraid cache
- thumbs
- encoded video
- everything else

I know it's possible when setting up your own docker container but wondering if it's still possible when using the community app here

**Comment by hydazz** [comment]
Yes, you can just map each folder to a different location on the host
so `/photos/thumbs` mounts to `/mnt/cache/immich/thumbs`
`/photos/library` mounts to `/mnt/x/immich/thumbs`
and so on...

**Comment by stevetsanders** [comment]
Ah gotcha, so adding those additional subdirectories would be done via adding another path?

<img width="825" alt="image" src="https://github.com/user-attachments/assets/8de1c341-fca2-417c-a776-6169c077024c">

**Comment by dilirity** [comment]
Wondering the same thing! Is such a thing possible with the unraid community image?

I couldn't find a variable that's used for the thumbnails directory.

**Comment by hydazz** [comment]
you'll want to try to avoid mounting a mount inside a mount, ie above you have `/mnt/user/photos_immich` mounted to `/photos`, but you also have `/mnt/user/appdata/immich/thumbs` mounted inside `/photos` - not the end of the world, but may cause issues

**Comment by dilirity** [comment]
I see, thanks.

What's the proper way of doing it?

**Comment by martabal** [comment]
You need to add these volumes (assuming you are using the storage template):
- `/mnt/user/photos_immich` &rarr; `/photos/library`
- `/mnt/user/appdata/immich/thumbs` &rarr; `/photos/thumbs`
- `/mnt/user/appdata/immich/profile` &rarr; `/photos/profile`
- `/mnt/user/appdata/immich/encoded-video` &rarr; `/photos/encoded-video`
- `/mnt/user/appdata/immich/upload` &rarr; `/photos/upload`

based on https://immich.app/docs/administration/backup-and-restore#filesystem

**Comment by dilirity** [comment]
I am not using the storage template.

**Comment by martabal** [comment]
> I am not using the storage template.

If you're not using the storage template, then you need to change that volume:

- `upload`: `/mnt/photos_immich` &rarr; `/photos/upload`

and remove `/photos/library`

**Comment by dilirity** [comment]
Awesome! Thanks @martabal!

Here's the updated mount list in case anyone needs to do a similar setup:

`VARIABLE NAME`: `Host Path` -> `Container Path`

- `Path: /photos:`: `/mnt/user/photos_immich/` -> `/photos/upload`
- `Path: /libraries:`: `/mnt/user/pictures/` -> `/libraries`
- `thumbnails`: `/mnt/user/appdata/immich/thumbs` -> `/photos/thumbs`
- `encoded-video`: `/mnt/user/appdata/immich/encoded-video` -> `/photos/encoded-video`
- `profile`: `/mnt/user/appdata/immich/profile` -> `/photos/profile`

**Comment by mustafababil** [comment]
> You need to add these volumes (assuming you are using the storage template):
> 
> * `/mnt/user/photos_immich` → `/photos/library`
> * `/mnt/user/appdata/immich/thumbs` → `/photos/thumbs`
> * `/mnt/user/appdata/immich/profile` → `/photos/profile`
> * `/mnt/user/appdata/immich/encoded-video` → `/photos/encoded-video`
> * `/mnt/user/appdata/immich/upload` → `/photos/upload`
> 
> based on https://immich.app/docs/administration/backup-and-restore#filesystem

Thanks for the guidance. 
I also had to specify backup folder as of v2.4.1
e.g. `/mnt/user/appdata/immich/backups` → `/photos/backups`.

Another hint;
if you see errors about reading `.immich` file in any of these folders, run `chown -R abc:users /photos/<folderName>` and restart the container.
